exporters/stdout/stdoutlog/exporter.go

```diff
@@ -33,24 +55,115 @@ func New(options ...Option) (*Exporter, error) {
 		timestamps: cfg.Timestamps,
 	}
 	e.encoder.Store(enc)
+	e.selfObservability = newSelfObservability()
 
 	return &e, nil
 }
 
+func newSelfObservability() *selfObservability {
+	if !x.SelfObservability.Enabled() {
+		return nil
+	}
+
+	selfObservability := &selfObservability{
+		enabled: true,
+		attrs: []attribute.KeyValue{
+			semconv.OTelComponentName(fmt.Sprintf("%s/%d", otelComponentType, nextExporterID())),
+			semconv.OTelComponentTypeKey.String(otelComponentType),
+		},
+	}
+
+	mp := otel.GetMeterProvider()
+	m := mp.Meter("go.opentelemetry.io/otel/exporters/stdout/stdoutlog",
+		metric.WithInstrumentationVersion(sdk.Version()),
+		metric.WithSchemaURL(semconv.SchemaURL),
+	)
+
+	var err error
+	if selfObservability.inflightMetric, err = otelconv.NewSDKExporterLogInflight(m); err != nil {
+		otel.Handle(err)
+	}
+	if selfObservability.exportedMetric, err = otelconv.NewSDKExporterLogExported(m); err != nil {
+		otel.Handle(err)
+	}
+	if selfObservability.operationDurationMetric, err = otelconv.NewSDKExporterOperationDuration(m); err != nil {
+		otel.Handle(err)
+	}
+
+	return selfObservability
+}
+
 // Export exports log records to writer.
 func (e *Exporter) Export(ctx context.Context, records []log.Record) error {
 	enc := e.encoder.Load()
 	if enc == nil {
 		return nil
 	}
 
+	var err error
+	if e.selfObservability != nil && e.selfObservability.enabled {
+		err = e.exportWithSelfObservability(ctx, records)
+	} else {
+		err = e.exportWithoutSelfObservability(ctx, records)
+	}
+	return err
+}
+
+const bufferSize = 1024
+
+var selfObservabilityBuffer = sync.Pool{
+	New: func() any {
+		buf := make([]attribute.KeyValue, 0, bufferSize)
+		return &buf
+	},
+}
+
+// exportWithSelfObservability exports logs with self-observability metrics.
+func (e *Exporter) exportWithSelfObservability(ctx context.Context, records []log.Record) (err error) {
+	count := int64(len(records))
+	start := time.Now()
+
+	e.selfObservability.inflightMetric.Add(ctx, count, e.selfObservability.attrs...)
+
+	defer func() {
+		bufPtrAny := selfObservabilityBuffer.Get()
+		bufPtr, ok := bufPtrAny.(*[]attribute.KeyValue)
+		if !ok || bufPtr == nil {
+			bufPtr = &[]attribute.KeyValue{}
+		}
+
+		addAttrs := (*bufPtr)[:0]
+		addAttrs = append(addAttrs, e.selfObservability.attrs...)
+
+		if err != nil {
+			addAttrs = append(addAttrs, semconv.ErrorType(err))
+		} else {
+			e.selfObservability.exportedMetric.Add(ctx, count, addAttrs...)
+		}
+
+		e.selfObservability.inflightMetric.Add(ctx, -count, e.selfObservability.attrs...)
+		e.selfObservability.operationDurationMetric.Record(ctx, time.Since(start).Seconds(), addAttrs...)
+
+		*bufPtr = addAttrs[:0]
+		selfObservabilityBuffer.Put(bufPtr)
```

Comment:
This needs to be `defer`ed right after the `Get` to ensure it will always be return to the pool.